### PR TITLE
[move-2024] Update sui framework tests to Move 2024

### PR DESCRIPTION
**Pull request by cgswords**
## Description 

This largely converts the sui framework tests to move 2024 syntax.

## Test Plan 

Lots of hand-review, plus the tests still work. In addition, I used local bytecode diffing.

Note that the following files have bytecode changes and should be reviewed with additional care:

- authenticator_state_tests.move
- balance_tests.move
- coin_tests.move
- ecdsa_k1_tests.move
- pay_tests.move
- random_tests.move
- test_scenario_tests.move
- verifier_tests.move
- zklogin_verified_id_tests.move
- zklogin_verified_issuer_tests.move

---
If your changes are not user-facing and do not break anything, you can skip the following section. Otherwise, please briefly describe what has changed under the Release Notes section.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: crates/sui-framework/packages/sui-framework/tests/authenticator_state_tests.move [hunk 1]
```diff
@@ -5,10 +5,9 @@
 #[test_only]
```
Comment (cgswords) [inline crates/sui-framework/packages/sui-framework/tests/authenticator_state_tests.move:5]: file changed -- review with care

File: crates/sui-framework/packages/sui-framework/tests/coin_tests.move [hunk 2]
```diff
@@ -3,15 +3,10 @@
```
Comment (cgswords) [inline crates/sui-framework/packages/sui-framework/tests/coin_tests.move:3]: file changed -- review with care

File: crates/sui-framework/packages/sui-framework/tests/crypto/ecdsa_k1_tests.move [hunk 3]
```diff
@@ -4,7 +4,6 @@
 #[test_only]
```
Comment (cgswords) [inline crates/sui-framework/packages/sui-framework/tests/crypto/ecdsa_k1_tests.move:4]: file changed -- review with care

File: crates/sui-framework/packages/sui-framework/tests/pay_tests.move [hunk 4]
```diff
@@ -3,8 +3,7 @@
```
Comment (cgswords) [inline crates/sui-framework/packages/sui-framework/tests/pay_tests.move:3]: file changed -- review with care

File: crates/sui-framework/packages/sui-framework/tests/verifier_tests.move [hunk 5]
```diff
@@ -15,11 +15,10 @@ module sui::verifier_tests {
         use sui::test_scenario;
         let admin = @0xBABE;
```
Comment (cgswords) [inline crates/sui-framework/packages/sui-framework/tests/verifier_tests.move:16]: file changed -- review with care

File: crates/sui-framework/packages/sui-framework/tests/crypto/zklogin_verified_issuer_tests.move [hunk 6]
```diff
@@ -4,13 +4,12 @@
 #[test_only]
```
Comment (cgswords) [inline crates/sui-framework/packages/sui-framework/tests/crypto/zklogin_verified_issuer_tests.move:4]: file changed -- review with care

File: crates/sui-framework/packages/sui-framework/tests/crypto/zklogin_verified_id_tests.move [hunk 7]
```diff
@@ -4,17 +4,16 @@
 #[test_only]
```
Comment (cgswords) [inline crates/sui-framework/packages/sui-framework/tests/crypto/zklogin_verified_id_tests.move:4]: file changed -- review with care

File: crates/sui-framework/packages/sui-framework/tests/random_tests.move [hunk 8]
```diff
@@ -4,43 +4,32 @@
 #[test_only]
```
Comment (cgswords) [inline crates/sui-framework/packages/sui-framework/tests/random_tests.move:4]: file changed -- review with care

File: crates/sui-framework/packages/sui-framework/tests/balance_tests.move [hunk 9]
```diff
@@ -3,7 +3,7 @@
```
Comment (cgswords) [inline crates/sui-framework/packages/sui-framework/tests/balance_tests.move:3]: file changed -- review with care

File: crates/sui-framework/packages/sui-framework/tests/crypto/zklogin_verified_id_tests.move [hunk 10]
```diff
@@ -4,17 +4,16 @@
...
     fun test_check_zklogin_id() {
         let address = @0x1c6b623a2f2c91333df730c98d220f11484953b391a3818680f922c264cc0c6b;
-        let kc_name = utf8(b"sub");
-        let kc_value = utf8(b"106294049240999307923");
-        let aud = utf8(b"575519204237-msop9ep45u2uo98hapqmngv8d84qdc8k.apps.googleusercontent.com");
-        let iss = utf8(b"https://accounts.google.com");
+        let kc_name = b"sub".to_string();
+        let kc_value = b"106294049240999307923".to_string();
+        let aud = b"575519204237-msop9ep45u2uo98hapqmngv8d84qdc8k.apps.googleusercontent.com".to_string();
+        let iss = b"https://accounts.google.com".to_string();
```
Comment (tzakian) [inline crates/sui-framework/packages/sui-framework/tests/crypto/zklogin_verified_id_tests.move:16]: Just to make sure: `to_string` here is calling `utf8` correct?

**Review by tzakian — APPROVED** [review]
Looks great to me!

File: crates/sui-framework/packages/sui-framework/tests/crypto/zklogin_verified_id_tests.move (hunk 10 above)
Comment (cgswords) [reply crates/sui-framework/packages/sui-framework/tests/crypto/zklogin_verified_id_tests.move:16]: Yep!

**Pull request merged**